provProto.py

```diff
@@ -0,0 +1,184 @@
+            else:
+                cursor.execute('''
+INSERT INTO prv_Task(taskName) VALUES ("%s")''' % task.name)
+                taskId = cursor.lastrowid
+                cursor.execute('''
+INSERT INTO prv_cnf_Task(taskId, validityBegin, validityEnd, gitSHA)
+VALUES (%d, "%s", "%s", "%s")''' % (taskId, self._currentTime, self._infinity,
+                                    task.gitSHA))
+                taskCnfId = cursor.lastrowid
+                for k in task.paramKVDict:
+                    cursor.execute('''
+INSERT INTO prv_cnf_Task_KVParams(taskCnfId, theKey, theValue)
+VALUES (%i, "%s", "%s")''' % (taskCnfId, k, task.paramKVDict[k]))
+            cursor.execute('''
+INSERT INTO prv_cnf_Pipeline_Tasks(pipelineCnfId, taskId, taskPosition)
+VALUES ("%s", "%s", %d)''' % (pipeCnfId, taskId, taskPos))
+            taskPos += 1
+
+        # register tables pipeline produces
+        for table in tables:
+            cursor.execute('''
+INSERT INTO prv_TableToPipeline(tableName, pipelineId) VALUES ("%s", "%s")''' % \
+                               (table, pipeId))
+            tToPId = cursor.lastrowid
+            cursor.execute('''
+INSERT INTO prv_cnf_TableToPipeline(tableToPipelineId, validityBegin,
+validityEnd) VALUES ("%s", "%s", "%s")''' % \
+                    (tToPId, self._currentTime, self._infinity))
+
+    def registerNode(self, cursor, name, ip, os, cores, ram):
+        """
+        Registers processing node in provenance.
+
+        @param name    Node name
+        @param ip      IP address of the node
+        @param os      OS name and version
+        @param cores   Number of cores
+        @param ram     Memory in GB
+        """
+        cursor.execute('INSERT INTO prv_Node(nodeName) VALUES ("%s")' % \
+                             (name))
+        nodeId = cursor.lastrowid
+        cursor.execute('''
+INSERT INTO prv_cnf_Node(nodeId, validityBegin, validityEnd, ip, os, cores, ram)
+VALUES ("%s", "%s", "%s", "%s", "%s", "%s", "%s")''' % \
+                (nodeId, self._currentTime, self._infinity, ip, os, cores, ram))
+
+    def updateTaskConfig(self, cursor, task):
+        """
+        Finds the configuration for a given task.
+
+        @param task       Task object. The names should point to an existing task
+                          The values provided will be used as the new values.
+        """
+        cursor.execute('''
+SELECT t.taskId FROM prv_cnf_Task ct, prv_Task t
+WHERE  ct.taskId = t.taskId
+  AND  t.taskName = "%s"
+  AND  validityEnd="%s"''' % (task.name, self._infinity))
+        taskId = cursor.fetchone()[0]
+        cursor.execute('''
+UPDATE prv_cnf_Task SET validityEnd="%s" WHERE taskId="%s"''' % \
+                           (self._currentTime, taskId))
+        cursor.execute('''
+INSERT INTO prv_cnf_Task(taskId, validityBegin, validityEnd, gitSHA)
+VALUES (%d, "%s", "%s", "%s")''' % \
+                (taskId, self._currentTime, self._infinity, task.gitSHA))
```

Comment:
It is not related to task execution. Task execution has it's own datetime. prv_cnf_Task is keeping track of different configurations of prv_Task. So whatever things we are tracking per task go here. I added gitSHA as a placeholder for now. The idea is that when we have a new version of algorithm that runs as part of a given task, then it triggers a new config for that task, and that new config knows how to find that algorithm